### DISCUSSION TITLE
Data received - how to know what it is

### DISCUSSION THREAD

**Discussion by promy** [codetheweb/tuyapi#507] [Q&A] [answered]
I was able to connect to my heat-pump. 
However, I'm not sure what parameters i am receiving.
I was hoping to get the data i can consult on the interface of the heat-pump but it seems that its something different.
I'm Receiving:
```
'1': true,
     '2': 24,
     '3': 24,
     '4': 'hot',
     '6': 0,
     '101': 1,
     '102': 16,
     '103': 15,
     '104': 19,
     '105': 17,
     '106': 0,
     '107': 19,
     '108': 24,
     '109': 24,
     '110': 0,
     '111': 0,
     '112': false,
     '113': false,
     '114': false,
     '115': false,
     '116': false,
     '117': false,
     '118': false
```
1 is the on/off ; 2 and 3 could be the temperature and 4 is the mode. Is there any way of knowing what every parameter is? Can I get more (ex. on the screen the current temps are 1 digit after the , (ex 23,5°C))?

**Comment by codetheweb** [comment, marked as answer]
Unfortunately it's very possible that the data accessible locally from your device is different from what the device sends to Tuya's cloud for consumption via the app.

There's not really a good way of listing parameters by type, best method is trial and error.

**Reply by promy** [reply]
Just wondering, if the device would send a float (1 digit after het ,), would it be kept in your app or would it be forced to an int?

**Reply by kueblc** [reply]
The `dps` object is decoded directly from a JSON string. No transformation is made beyond that, ie no casting of floats to ints. The code you referenced deals with decoding the underlying transport, not the device data.

In your case, the missing decimal of precision might be in another `dp`. I have seen cases where precision is to 0.5, the integer part being in one `dp`, and the 0.5 represented by a 0 or 1 in another `dp`.
